alicloud/resource_alicloud_cms_sitemonitor.go

```diff
@@ -0,0 +1,193 @@
+package alicloud
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/cms"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudCmsSiteMonitor() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudCmsSiteMonitorCreate,
+		Read:   resourceAlicloudCmsSiteMonitorRead,
+		Update: resourceAlicloudCmsSiteMonitorUpdate,
+		Delete: resourceAlicloudCmsSiteMonitorDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"address": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"task_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"task_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{SiteMonitorHTTP, SiteMonitorDNS, SiteMonitorFTP, SiteMonitorPOP3,
+					SiteMonitorPing, SiteMonitorSMTP, SiteMonitorTCP, SiteMonitorUDP}, false),
+			},
+			"alert_ids": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"interval": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntInSlice([]int{1, 5, 15}),
+			},
+			"isp_cities": {
+				Type:     schema.TypeString,
```

Comment:
still applies

Reply:
It is defined and imlemented as simple string https://help.aliyun.com/document_detail/115048.html
Format is serialized json object.

Reply:
still applies

Reply:
since `isp_cities` is not needed by customer currently, I'll remove and add at a later iteration to speed up merge.